### PR TITLE
Api/connect survey submission

### PR DESCRIPTION
**Pull request by ElijahRW**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
      * Connected Submission service to API Endpoint.
      * Added ths_sectionc_example to the submission.

### CODE REVIEW

File: api-objects/AppointmentSubmission.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+export class AppointmentSubmission {
+    constructor(clone?: AppointmentSubmission) {
+        if (clone) {
+            this.patient = clone.patient;
+            this.patientSurvey = clone.patientSurvey;
+        }
+    }
+
+    patientSurvey: PatientSurveyJSON;
+    patient: PatientJSON;
```
Comment (gstone271) [inline api-objects/AppointmentSubmission.ts]: access modifier?

File: api-objects/AppointmentSubmission.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    public exportToSQLVars() {
```
Comment (gstone271) [inline api-objects/AppointmentSubmission.ts:12]: this is one of the "shitty translation layers" that the guy who came in to our meeting, I forgot his name, was talking about.

Making it unnecessary might be more work than it's worth though.

File: api/appointmentsPOST.ts [hunk 3]
```diff
@@ -47,40 +49,55 @@ let appointment_datapoints : string [] = [
...
-  let tssurvey_values = ts_datapoints.map((value, index) => request.body[value])
+  let tssurvey_values = ts_datapoints.map((value) => body[value])
 
   let audiologistexam_sql : string = "INSERT INTO audiologistexams (" + audiologistexams_datapoints.join() + ") VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING *";
-  let audiologistexam_values = audiologistexams_datapoints.map((value, index) => request.body[value])
+  let audiologistexam_values = audiologistexams_datapoints.map((value) => body[value]);
 
   // Appointments inserted last because it needs info from the other tables
-    await Promise.all([
+    return await Promise.all([
```
Comment (gstone271) [inline api/appointmentsPOST.ts]: use `async` style by replacing this with `const values = await Promise....`

File: src/app/audiologist-navigation/audiologist-navigation.component.ts [hunk 4]
```diff
@@ -47,7 +48,18 @@ export class AudiologistNavigationComponent {
...
-      this.summaryComponent.submitSurvey();
+      this.summaryComponent.submitSurvey().subscribe(
+        _ => {
+          alert('Survey was Sucessfully Submitted');
+          this.logout();
+        },
+        catchError(
+          (error, _) => {
+            alert('Survey Failed to Submit');
+            throw error;
```
Comment (gstone271) [inline src/app/audiologist-navigation/audiologist-navigation.component.ts]: Have you tested this case? Where is this error caught? Does the whole application explode?

File: api-objects/AppointmentSubmission.ts (hunk 2 above)
Comment (ElijahRW) [reply api-objects/AppointmentSubmission.ts:12]: Lol, YUP. Luke _(That was Luke)_ and I both have had unpleasant experiences with connectors like these. My hope is that we will have some time during **week 9** to fix some of these connection breaks: The front end could benefit by using this class.

File: api/appointmentsPOST.ts (hunk 3 above)
Comment (ElijahRW) [reply api/appointmentsPOST.ts]: As in rewrite this line to 
`const values = await Promise.all([`
then add a line after the catch:
`return values;`
Right?

File: src/app/audiologist-navigation/audiologist-navigation.component.ts (hunk 4 above)
Comment (ElijahRW) [reply src/app/audiologist-navigation/audiologist-navigation.component.ts]: The error is caught by Angular's default error handler which displays the message to Chrome's console so the application remains stable. However, you are right, its NOT pretty. I'll remove this catchError here entirely. (This should be handled by an error service: _See [this branch](https://github.com/movshov/VA-Audiology-App/tree/CLIENT/ErrorHandler) to see what I hope to pull in after the demo.)_

**Review by Liriaene — APPROVED** [review]
Changes look good to me. 
As a side question, is this the new direction the endpoints are going to take?

File: api/appointmentsPOST.ts (hunk 3 above)
Comment (gstone271) [reply api/appointmentsPOST.ts]: No. `async` style eliminates the call to `then`. What makes `async` functions easier to read is that they are not written with calls to `then` (even though they are secretly implemented with them). 

When you write `const foo = await bar; const foo1 = foo + 1;`, this is translated to `bar.then(foo => { const foo1 = foo + 1; }`

`values` here is the same variable as (and eliminates) the function parameter `values` to the lambda. Eliminate the usage of this lambda.

File: src/app/audiologist-navigation/audiologist-navigation.component.ts [hunk 5]
```diff
@@ -50,7 +51,13 @@ export class AudiologistNavigationComponent {
 
   public submitSurvey() {
     if (this.summaryComponent != null) {
-      this.summaryComponent.submitSurvey();
+      this.summaryComponent.submitSurvey().subscribe(
+        _ => {
+          // TODO: Replace Alert With Notification call once it is pulled in.
```
Comment (OkapalDominic) [inline src/app/audiologist-navigation/audiologist-navigation.component.ts]: It's pulled in.

**Review by MishalAlajmi-psu — APPROVED** [review]
Looks good.

**Review by OkapalDominic — APPROVED** [review]
I approve.

**Pull request merged**